Content.Server/Automod/AutomodManager.cs

```diff
@@ -0,0 +1,243 @@
+                textMatches.Add((str, match.Index));
+            }
+
+            if (!_protoMan.TryIndex<AutomodActionGroupPrototype>(filter.ActionGroup, out var censorGroup))
+            {
+                _log.Error($"AutomodActionGroupPrototype \"{filter.ActionGroup}\" not found.");
+                continue;
+            }
+
+            var skip = false;
+            foreach (var censorAction in censorGroup.AutomodActions)
+            {
+                if (!censorAction.Skip(inputText, textMatches))
+                    continue;
+
+                skip = true;
+                break;
+            }
+            if (skip)
+                continue;
+
+            var displayName = GetDisplayName(filter);
+
+            foreach (var censorAction in censorGroup.AutomodActions)
+            {
+                passes &= censorAction.RunAction(session, inputText, textMatches, filter, displayName, _entMan);
+            }
+        }
+
+        return passes;
+    }
+
+    private string GetDisplayName(AutomodFilterDef filter)
+    {
+        return filter.Id != null ? $"{filter.DisplayName} (automod#{filter.Id})" : $"{filter.DisplayName} (automod#na)";
+    }
+
+    #endregion
+
+    #region Reload
+
+    public async void ReloadAutomodFilters()
+    {
+        _regexFilters.Clear();
+
+        var filters = await _db.GetAllAutomodFiltersAsync();
+
+        foreach (var filter in filters)
+        {
+            AddFilter(filter);
+        }
+    }
+
+    #endregion
+
+    #region Create
+
+    public async void CreateFilter(string pattern,
+        AutomodFilterType filterType,
+        string actionGroup,
+        AutomodTarget targets,
+        string name)
+    {
+        CreateFilter(new AutomodFilterDef(pattern, filterType, actionGroup, targets, name));
+    }
+
+    public async void CreateFilter(AutomodFilterDef automod)
+    {
+        automod = await _db.AddAutomodFilterAsync(automod);
+        AddFilter(automod);
+    }
+
+    /// <summary>
+    /// Adds a filter to the manager lists.
+    /// </summary>
+    /// <param name="automod"></param>
+    private void AddFilter(AutomodFilterDef automod)
+    {
+        foreach (var targetFlag in Enum.GetValues<AutomodTarget>())
+        {
+            if (targetFlag == AutomodTarget.None)
+                continue;
+            if (!automod.TargetFlags.HasFlag(targetFlag))
+                continue;
+
+            if (automod.FilterType == AutomodFilterType.Regex)
+            {
+                var list = _regexFilters.GetOrNew(targetFlag);
+                _regexFilters.TryAdd(targetFlag, list);
+
+                try
+                {
+                    var regex = new Regex(automod.Pattern);
+
+                    list.Add(new RegexAutomodFilterDef(automod, regex));
+                }
+                catch (RegexParseException e)
+                {
+                    _log.Error("Failed to parse regex in {0}. Error code: {1}.", this.GetType(), e.Error);
```

Comment:
This should really say *which* filter (you have IDs/names) is failing to parse. And I'm not sure the utility of logging `this.GetType()`.

Reply:
Changed in [1056fa7](https://github.com/space-wizards/space-station-14/pull/28570/commits/1056fa780d3de9d5984195a6bd0e6eec0bf66d53)